lib/quickbooks/model/active_record_scaffold.rb

```diff
@@ -0,0 +1,85 @@
+module Quickbooks
+  module Model
+    module ActiveRecordScaffold
+
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        def all(query = nil, options = {})
+          initialize_service(options).query(query, options).entries
+        end
+
+        def where(options = {})
+          if options.is_a?(String)
+            all(build_string_query(options), {})
+          elsif options.present?
+            all(build_hash_query(options.except(:skip_pagination)), options)
+          else
+            all(full_query, {})
+          end
+        end
+
+        def find_by(options = {})
+          where(options).first
+        end
+
+        def find(id)
+          find_by(id: id)
+        end
+
+        def count
+          initialize_service.query("SELECT COUNT(*) FROM #{model}").total_count
+        end
+
+        def query_in_batches(options = {})
+          initialize_service(options).query_in_batches(query, options).entries
+        end
+
+        private
+
+        def initialize_service(options = {})
+          service              = eval("Quickbooks::Service::#{model}").new
+          service.access_token = OAuth::AccessToken.new(
+            $qb_oauth_consumer, options[:token] || Thread.current[:token],
```

Comment:
@kashif-umair where does `$qb_oauth_consumer` come from? This indicates its a global variable which needs to be set by the user. We'll need a better way to load this instance of `OAuth::Consumer` into the library.

I'm going to un-do this PR until this can be resolved as its currently in a non-functional state.